### PR TITLE
Create sample creation date

### PR DESCRIPTION
**Pull request by jleblond**
Thurs Feb. 25, 2021: To be reviewed

can be squashed and merged when the PR for the frontend will be approved

https://github.com/c3g/freezeman_web/pull/47

### CODE REVIEW

**Comment by jleblond** [comment]
> Quick comment ... There should be only one migration by release. If 11 was 3.0, 12 should be all of 3.1.

I need the data from the reception_date to fill the creation_date column, so the order of the migrations is important.

**Comment by davidlougheed** [comment]
operations should be executed in order

**Comment by davidlougheed** [comment]
sorry, misclick

**Comment by UlysseFG** [comment]
Yeah as David mentioned each operation in a single migration is executed in order so you can just append your other migration operations after the ones in the previous migration.

**Comment by jleblond** [comment]
Not from my tests, and from what I am finding on the Internet.. the order is not respected, therefore I get an error during the migration because the field reception_date does not exist

**Comment by UlysseFG** [comment]
This is likely because the model you use to process the migration no longer has a reception_date in it. If you do the copying using an SQL query this should work. You can check in the previous migrations (0008_... for example). They include field copies that were made in a precise order.

**Comment by davidlougheed** [comment]
I actually didn't mean to comment at all but I guess hit "close with comment" by accident with something I wasn't going to send in that form at least (sorry!)

There's also https://github.com/c3g/freezeman_server/blob/master/fms_core/migrations/0005_v2_0_0.py which did some field re-naming/shuffling stuff, not sure if anything I did there would be of help

File: fms_core/resources/sample.py [hunk 1]
```diff
@@ -37,7 +37,7 @@ class SampleResource(GenericResource):
     tissue_source = Field(attribute='tissue_source', column_name='Tissue Source')
     concentration = Field(attribute='concentration', column_name='Conc. (ng/uL)', widget=DecimalWidget())
     depleted = Field(attribute='depleted', column_name='Source Depleted')
-    reception_date = Field(attribute='reception_date', column_name='Reception Date', widget=DateWidget())
+    creation_date = Field(attribute='creation_date', column_name='Creation Date', widget=DateWidget())
```
Comment (UlysseFG) [inline fms_core/resources/sample.py:40]: Je ne changerais pas le column_name ici (surtout si tu ne changes pas le column_name dans le template) Le user doit savoir qu'il soumet un reception date.

File: fms_core/migrations/0012_v3_1_0.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('fms_core', '0011_v3_0_0'),
+    ]
+
+    operations = [
+        migrations.RenameField(
```
Comment (UlysseFG) [inline fms_core/migrations/0012_v3_1_0.py:14]: J'aime bien cette solution ; ) En passant, je ne sais pas si tu as d'autre changements qui vont s'ajouter, mais la creation d'un creation_date devrait inclure l'ajout d'une colonne dans le template (et la ressource) extraction pour ajouter un date_extracted. Faudrait confirmer avec David, mais l'ajout de samples a Freezeman ne se fait pas nécessairement la journée même d'une opération alors je permettrait que la date soit différente de la date courante de l'extraction dans freezeman...

**Review by UlysseFG — CHANGES_REQUESTED** [review]
This looks good, you have not renamed the samples versions fields from reception_date to creation_date.

**Review by UlysseFG — APPROVED** [review]
Looks good

**Pull request merged**